src/movement_controls.py

```diff
@@ -0,0 +1,267 @@
+
+        # a stationary car cant turn
+        self.current_turnVelocity = input.rx_axis.value * self.turn_speed * delta_time 
+
+
+    def ghost_screen_facing(self,delta_time: float, input: PlayerInput, vehicle_sprite: arcade.Sprite):
+        # ghost controls but agnostic of the current facing of the car
+
+
+        acceleration_change = 0
+        if input.x_axis.value != 0 or input.y_axis.value != 0 :
+            acceleration_change += self.acceleration_rate
+        else:
+            if abs(self.current_acceleration) != 0:
+                if abs(self.current_acceleration) <= delta_time * self.friction:
+                    self.current_acceleration = 0
+                else:
+                    acceleration_change -= delta_time * self.friction
+        
+        # the current acceleration is used to have speed progress from zero to its maximum 
+        self.current_acceleration = self.clamp(self.current_acceleration + (acceleration_change * delta_time),0,1)
+
+        # NOTE: because of the place holder sprite's facing add 90 degrees
+        car_angle = math.radians(vehicle_sprite.angle )
+
+
+        if abs(input.y_axis.value) + abs(input.x_axis.value) > 1:
+            xNormalized = input.x_axis.value / (input.y_axis.value + input.x_axis.value)
+            yNormalized = input.y_axis.value / (input.y_axis.value + input.x_axis.value)
+        else:
+            xNormalized =  self.drive_speed * input.x_axis.value * self.current_acceleration * delta_time 
+            yNormalized =  self.drive_speed * input.y_axis.value * self.current_acceleration * delta_time
+
+
+        self.current_velocity_y =  self.drive_speed * yNormalized * self.current_acceleration * delta_time
+        self.current_velocity_x =  self.drive_speed * xNormalized * self.current_acceleration * delta_time  
+        
+        self.current_turnVelocity  = input.rx_axis.value * self.turn_speed * delta_time 
+
+    def mars_walker(self,delta_time: float, input: PlayerInput,vehicle_sprite: arcade.Sprite):
+        # speed progressing from 0 to max while either acceleration is held
+        # otherwise the car should coast to a stop using the friction value [TODO:that should be a constant]
+
+
+        if input.x_axis.value != 0:
+            self.current_velocity_x = 0
+            self.current_velocity_y = 0
+            self.current_turnVelocity  = input.x_axis.value * self.turn_speed * delta_time 
+        else:
+            self.current_turnVelocity  = 0
+            car_angle = math.radians(vehicle_sprite.angle)
+
+            # find the X and Y movement based on the angle of the sprite
+            self.current_velocity_x =  self.drive_speed * math.cos(car_angle) * input.accelerate_axis.value * delta_time
+            self.current_velocity_y =  self.drive_speed * math.sin(car_angle) * input.accelerate_axis.value * delta_time
+
+
+
+
+
+
+    def clamp(self,num, min_value, max_value):
+        return max(min(num, max_value), min_value)
```

Comment:
Should we move `clamp` into `iron_math.py`?

Reply:
would be a question for collisions since that is the other place it might be used